sdr/concepts_and_interactions.md

```diff
@@ -0,0 +1,202 @@
+  * Moab manipultion and comparison code mostly lives in the [moab-versioning gem](https://github.com/sul-dlss/moab-versioning/), but also some in preservation_robots and preservation_catalog.
+* How things go in
+  * a [preservation_robots](https://github.com/sul-dlss/preservation_robots/) worker takes a BagIt bag containing data and metadata for a new object version, and uses that to create a new Moab (if v1 of the object) or add a new version to an existing Moab (if v2+ of the object).
+  * a preservation_robots worker will also ask preservation_catalog to queue an integrity check of the new Moab version.
+* Moabs are stored on "storage roots".  There are currently 3 in production.
+  * new content goes to the latest storage root (as configured in the storage root list given to the moab-versioning gem)
+  * updates to existing content go to the Moab directory on its current storage root (which may no longer be the last storage root)
+  * Moabs are stored in "druid tree" paths, which are similar to pair tree paths (e.g. druid `bc123df4567` would be `bc/123/df/4567/bc123df4567`)
+* What are the APIs to interact with preservation
+  * preservation-client is a ruby gem for interating with pres cat's REST API
+* Once preservation_catalog has been made aware that new object has been preserved or a new version has been preserved for an existing object, assuming the object passes structural and checksum validation, preservation catalog will replicate the new object version to the cloud.
+  * It creates a zip file (uncompressed) of the new Moab version, and sends that to each of the S3 endpoints (there are currently 3, 2 AWS and 1 IBM).
+* How is redis used in preservation
+  * there is a Redis instance that coordinates all of the regular accessioning "robots" (including preservation_robots)
+  * there is a Redis instance that coordinates all of the preservation_catalog workers (for replicating copies to the cloud and for auditing all preserved content both on prem in in the cloud)
+* The preservation catalog codebase contains a database README with an ER diagram for the schema as well as a number of useful example queries, and a diagram and README for its replication worker pipeline.
+
+## Technical Metadata app
+_Service to extract and record technical metadata for files deposited into SDR_
+
+(ask JLitt)
+
+* technical metadata: for each file, its file type, characteristics of image, PDF, A/V files and so on
+
+## SDR Tags database
+_Accessioneers and Argo users and some of our apps "tag" digital objects in the SDR_
+
+(ask Mike)
+
+  * A table in dor-services-app's PostgreSQL database.  Used to live in Fedora.  Useful for human administration of repository content in Argo.
+
+## Modsulator
+_A gem and an app that can turn spreadsheets into MODS in a single bound ... or something_
+
+## Bulk jobs/actions in Argo
+_When making changes to objects one at a time is the wrong approach_
```

Comment:
This should probably go on an "Features of Argo" page.

Reply:
https://github.com/sul-dlss/infrastructure-team/issues/8